src/Processors/Formats/Impl/ParquetBlockInputFormat.cpp

```diff
@@ -238,6 +241,50 @@ static std::optional<Field> decodePlainParquetValueSlow(const std::string & data
     return field;
 }
 
+static std::size_t getHeaderIndexByColumnName(const std::string & column_name, const Block & header)
+{
+    for (std::size_t i = 0; i < header.columns(); ++i)
+    {
+        if (header.getByPosition(i).name == column_name)
+        {
+            return i;
+        }
+    }
+
+    throw Exception(ErrorCodes::BAD_ARGUMENTS, "Column {} not found in Parquet file", column_name);
+}
+
+static ParquetBloomFilterCondition::IndexToColumnBF buildColumnIndexToBF(
+    parquet::BloomFilterReader & bf_reader,
+    int row_group,
+    const Block & header,
+    const std::vector<std::pair<std::string, int>> & column_name_to_index
+)
+{
+    auto rg_bf = bf_reader.RowGroup(row_group);
+
+    if (!rg_bf)
+    {
+        return {};
+    }
+
+    ParquetBloomFilterCondition::IndexToColumnBF index_to_column_bf;
+
+    for (const auto & [column_name, index] : column_name_to_index)
+    {
+        auto bf = rg_bf->GetColumnBloomFilter(index);
```

Comment:
I am sorry, I didn't quite understand this last comment. What do you mean by unneeded indexes in this case?

In the example you have shown, why is reading bf for `x` unnecessary?

Reply:
Maybe github failed to show the parent comment? It was supposed to look like this:

![Screenshot from 2024-08-23 13-11-57](https://github.com/user-attachments/assets/8b935c61-27b9-4f2b-ac85-9cd97b115f5b)

E.g. for `SELECT ... WHERE x > 42`, there's no point reading bloom filter for `x`. Because the `ParquetBloomFilterCondition` is just FUNCTION_UNKNOWN with no mention of column `x`.

For `x = 1 or y > 2` the `ParquetBloomFilterCondition` is [`x == 1`, `FUNCTION_UNKNOWN`, `FUNCTION_OR`], which is equivalent to [`FUNCTION_UNKNOWN`]. The result of `x = 1` bloom filter check is effectively discarded. (But it can be made to work: the bloom filter check and min/max statistics check (see `checkInHyperrectangle` call in `ParquetBlockInputFormat`) can be merged into one, using one KeyCondition. So the `x = 1` RPNElement would use bf, the `y > 2` RPNElement would use min/max statistics, and the two can be meaningfully ORed together.)

Reply:
I might be missing something, but taking the union of all `key_columns` from ConditionElements in the case of `x = 1 or y > 2` would still contain `x` even tho it is not necessary as you mentioned.

If by skipping unneeded indexes you mean "only read bloom filter for columns that are present in the where predicate", this is already addressed by `filtering_columns`